src/main/java/net/sf/jabref/logic/exporter/ExportFormat.java

```diff
@@ -310,7 +324,7 @@ public void performExport(final BibDatabaseContext databaseContext, final String
             }
 
             // Clear custom name formatters:
-            Globals.prefs.customExportNameFormatters = null;
+            layoutPreferences.getCustomExportNameFormatters().clear();
```

Comment:
There should be a method offered at layoutPreferences to clear the list

Typically, a copy of the internal data structure should be returned and thus `clear()` has no effect. layoutPreferences.getCustomExportNameFormatters should be wrapped (there!) in [Collections.unmodifiableList](http://docs.oracle.com/javase/7/docs/api/java/util/Collections.html#unmodifiableList%28java.util.List%29).